broadinstitute/single_cell_portal_core
# Integrating HCA results from Azul into bulk download UX (SCP-3812, SCP-3815)

This update integrates the preceding work leveraging searches via Azul into the bulk download UX.  This also includes updates to our analytics reporting to Mixpanel to aid in distinguishing study- from search-based bulk download requests.  This update also includes refactoring that has improved latency on XDSS requests, as we no longer ask for file-based information as a part of a search request.  This logic has not been moved into the `BulkDownloadService`, as normal search responses contain summary-level information about files (including types, formats, counts, and total bytes) that are sufficient to use in the bulk download modal.

MANUAL TESTING
1. Pull this branch, run any remaining migrations via `bin/rails db:migrate`
2. Enter the Rails console and run `SearchFacet.update_all_facet_filters` (this is to ensure you have all filter values from Azul)
3. Boot as normal, sign in with an admin account, and make sure the `cross_dataset_search_backend` feature flag is turned on
4. On the home page, execute a search for `species:Homo sapiens` and `disease:HIV infectious disease`
5. Ensure the following HCA studies appear, and match correctly on both facets:
![Screen Shot 2021-11-18 at 3 38 40 PM](https://user-images.githubusercontent.com/729968/142494062-0264439d-8d6d-45f1-8117-4f7faf7c5924.png)
6. Click "Download", and select all files entries for the HCA project `An Atlas of Immune Cell Exhaustion in HIV-Infected Individuals Revealed by Single-Cell Transcriptomics`
7. Click through to "Authorize", and paste the curl command into a terminal window
8. Ensure that the request succeeds, and files begin downloading into a directory called `ImmuneCellExhaustianHIV` (you can hit `CTRL^C` once the download starts)
9. (Optional) View [this](https://mixpanel.com/project/2085496/view/19055/app/insights#report/26971107/cross-dataset-search-requests) report in Mixpanel (dev analytics) and took for your download request to see that the line for `ImmuneCellExhaustianHIV` with `135` for `numAzulFiles` has an entry (there may be more depending on other usage)
